pkg/util/url/url.go

```diff
@@ -0,0 +1,434 @@
+package url
+
+import (
+	"errors"
+	gurl "net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/readium/go-toolkit/pkg/internal/extensions"
+	"golang.org/x/net/idna"
+)
+
+/*
+A Uniform Resource Locator.
+
+https://url.spec.whatwg.org/
+*/
+type URL interface {
+	Path() string            // Decoded path segments identifying a location.
+	Filename() string        // Decoded filename portion of the URL path.
+	Extension() string       // Extension of the filename portion of the URL path.
+	Query() gurl.Values      // Returns the query parameters present in this URL.
```

Comment:
I would remove this API in favor of using straight `url.Raw().Query()` where needed for now. This way it is more future-proof if we need to add our own data structure representing query parameters, or if we switch to a different implementation of URL than `net/url`.

```suggestion
```